server/src/main/java/org/opensearch/action/search/AbstractSearchAsyncAction.java

```diff
@@ -441,7 +441,10 @@ private void onPhaseEnd(SearchRequestContext searchRequestContext) {
 
     private void onPhaseStart(SearchPhase phase) {
         setCurrentPhase(phase);
-        this.searchRequestContext.getSearchRequestOperationsListener().onPhaseStart(this);
+        try {
+            SearchPhaseName.valueOf(phase.getSearchPhaseName().toString());
+            this.searchRequestContext.getSearchRequestOperationsListener().onPhaseStart(this);
+        } catch (IllegalArgumentException ignored) {}
```

Comment:
Let's try to find another solution for this. 

There are a couple of downsides to this exception-throwing approach:
1. Filling in a stack trace can be quite expensive, especially with the rather deep stacks that we tend to have in OpenSearch.
2. If you throw a standard exception enough times, the JVM will eventually stop filling in stack traces (to avoid the expense called out above), but then it becomes harder to root-cause an exception.

Could we instead give it a name like `prefilter` and explicitly check for that? Alternatively, maybe add a static method to `SearchPhaseName` that checks for a valid phase name?

Reply:
I thought stack trace will not be printed with try catch. If logging is needed we can print something other than the actual error trace.

It is not a true search phase so we thought "none" is a better name. Also I saw there are other phases like create_pit  that we can afford to skip onPhaseStart() execution for since we only track these 6 search phases in coordinator node stats, phase_took, slog log.

Reply:
> I thought stack trace will not be printed with try catch. If logging is needed we can print something other than the actual error trace.

Oh, it's not about printing the stack trace, but rather computing the stack trace in the exception's constructor.

> It is not a true search phase so we thought "none" is a better name. Also I saw there are other phases like create_pit that we can afford to skip onPhaseStart() execution for since we only track these 6 search phases in coordinator node stats, phase_took, slog log.

`none` is an okay name in that case. 

Maybe in `SearchPhaseNames` you can add a little helper like:

```
    private static final Set<String> PHASE_NAMES = new HashSet<>();
    static {
        for (SearchPhaseName phaseName : SearchPhaseName.values()) {
            PHASE_NAMES.add(phaseName.name);
        }
    }
    public boolean isValidName(String phaseName) {
        return PHASE_NAMES.contains(phaseName);
    }
```